enterprise/security/src/main/java/org/slf4j/impl/StaticLoggerBinder.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2002-2016 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.slf4j.impl;
+
+import org.slf4j.ILoggerFactory;
+import org.slf4j.spi.LoggerFactoryBinder;
+
+import org.neo4j.logging.LogProvider;
+
+public class StaticLoggerBinder implements LoggerFactoryBinder {
+    private static final StaticLoggerBinder SINGLETON = new StaticLoggerBinder();
+    public static String REQUESTED_API_VERSION = "1.6.99";
+    private static final String loggerFactoryClassStr = Neo4jLoggerFactory.class.getName();
+    private final ILoggerFactory loggerFactory = new Neo4jLoggerFactory();
+
+    public static final StaticLoggerBinder getSingleton() {
+        return SINGLETON;
+    }
+
+    private StaticLoggerBinder() {
+    }
+
+    public ILoggerFactory getLoggerFactory() {
+        return this.loggerFactory;
+    }
+
+    public String getLoggerFactoryClassStr() {
+        return loggerFactoryClassStr;
+    }
+
+    public static void setNeo4jLogProvider( LogProvider logProvider )
+    {
+        ((Neo4jLoggerFactory) getSingleton().getLoggerFactory()).setNeo4jLogProvider( logProvider );
```

Comment:
If `getLoggerFactory()` always returns `Neo4jLoggerFactory`, why don't we change its return type, and skip having to cast here?